pkg/resource/v1/chartconfig/resource.go

```diff
@@ -14,22 +21,29 @@ const (
 
 // Config represents the configuration used to create a new chart config resource.
 type Config struct {
-	G8sClient   versioned.Interface
-	K8sClient   kubernetes.Interface
-	Logger      micrologger.Logger
-	ProjectName string
+	BaseClusterConfig        cluster.Config
+	G8sClient                versioned.Interface
+	K8sClient                kubernetes.Interface
+	Logger                   micrologger.Logger
+	ProjectName              string
+	ToClusterGuestConfigFunc func(obj interface{}) (v1alpha1.ClusterGuestConfig, error)
 }
 
 // Resource implements the chart config resource.
 type Resource struct {
-	g8sClient   versioned.Interface
-	k8sClient   kubernetes.Interface
-	logger      micrologger.Logger
-	projectName string
+	baseClusterConfig        cluster.Config
+	g8sClient                versioned.Interface
+	k8sClient                kubernetes.Interface
+	logger                   micrologger.Logger
+	projectName              string
+	toClusterGuestConfigFunc func(obj interface{}) (v1alpha1.ClusterGuestConfig, error)
 }
 
 // New creates a new configured chart config resource.
 func New(config Config) (*Resource, error) {
+	if reflect.DeepEqual(config.BaseClusterConfig, cluster.Config{}) {
```

Comment:
Lets please provide validate methods for these objects in the future. We have to hack validation in a rather poor way everywhere. I know you did not make this up but I noticed this right now. So something to keep in mind.